test/tools/libtesteth/Options.cpp

```diff
@@ -72,6 +74,26 @@ void printVersion()
 {
     cout << prepareVersionString() << "\n";
 }
+}
+
+void Options::setVerbosity(int _level)
+{
+    static std::ostringstream strCout;  // static string to redirect logs to
+    if (_level > verbosity)
```

Comment:
I think the code was / would be much simpler when you don't need this reentrance of `setVerbosity` support and call it just once in the end of `Options` constructor.

Reply:
then this if will be each time you set a verbosity

Reply:
You should `setVerbosity` only once